websites/N/NIKKE.GG/presence.ts

```diff
@@ -0,0 +1,89 @@
+const presence = new Presence({
+		clientId: "1230509717055602849",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets {
+	Logo = "https://i.imgur.com/nA6br84.png",
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+			startTimestamp: browsingTimestamp,
+		},
+		getImageCharacter = (url: string) =>
+			`${"https://static.dotgg.gg/nikke/characters/"}si_c${
+				url.match(/c(\d+)_\d+/)[1]
+			}_${url.match(/c\d+_(\d+)/)[1]}_s.webp`,
+		getImageMonster = (url: string) =>
+			`https://static.dotgg.gg/nikke/monsters/si_${url
+				.substring(url.lastIndexOf("/") + 1, url.lastIndexOf("."))
+				.replace("full_", "")}.webp`,
+		{ href, pathname, search } = document.location,
+		thumbnail = document
+			.querySelector<HTMLMetaElement>('meta[property="og:image"]')
+			?.getAttribute("content"),
+		title = document
+			.querySelector<HTMLMetaElement>('meta[property="og:title"]')
+			?.getAttribute("content");
+
+	if (search.startsWith("?s")) {
+		presenceData.largeImageKey = Assets.Logo;
+		presenceData.state = search.split("&")[0].split("=")[1].replace(/\+/g, " ");
+		presenceData.details = "Searching for";
+		presenceData.smallImageKey = Assets.Search;
+	} else if (pathname.startsWith("/characters")) {
+		if (pathname === "/characters/") {
+			presenceData.largeImageKey = Assets.Logo;
+			presenceData.details = "Viewing Characters List";
+			presenceData.smallImageKey = Assets.Viewing;
+		} else {
+			presenceData.largeImageKey = getImageCharacter(thumbnail);
+			presenceData.state = `${title.split("|")[0]}`;
+			presenceData.details = "Viewing Character";
+			presenceData.smallImageKey = Assets.Reading;
+			presenceData.smallImageText = "Reading";
+			presenceData.buttons = [{ label: "View Character", url: href }];
+		}
+	} else if (pathname.startsWith("/monsters")) {
+		if (pathname === "/monsters/") {
+			presenceData.largeImageKey = Assets.Logo;
+			presenceData.details = "Viewing Rapture List";
+			presenceData.smallImageKey = Assets.Viewing;
+		} else {
+			presenceData.largeImageKey = `${getImageMonster(
+				document
+					.querySelectorAll<HTMLMetaElement>('meta[property="og:image"]')[1]
+					.getAttribute("content")
+					.replace(".png.png", ".png")
+			)}`;
```

Comment:
This is simpler and works
```suggestion
			presenceData.largeImageKey = document.querySelector<HTMLImageElement>('.img > img')?.src ?? Assets.Logo
```

Reply:
That directly images not resizing properly for a thumbnail (need at least symmetrical), that's why came from another but still had relation